addon/lib/user.js

```diff
@@ -64,3 +68,129 @@ function makeUuid() {
 exports.getUserInfo = function () {
   return ss.storage.userInfo;
 };
+
+exports.updateProfileInfo = function (profile) {
+  if (! profile) {
+    return;
+  }
+  let currentInfo = ss.storage.profileInfo;
+  if (! currentInfo) {
+    currentInfo = ss.storage.profileInfo = {};
+  }
+  let attrs = Object.keys(profile);
+  for (let attr of attrs) {
```

Comment:
> Or are you worried about null values that don't mean erase...

Good catch! I overloaded `updateProfileInfo` to mean, "update this profile record with the user's changes," and "fill in default values from FxA if the user hasn't provided them yet," because I didn't know how I wanted them to fit together. This definitely calls for separate methods now, though.